OneShot docs (#1648)
PX4/PX4-user_guide: en/peripherals/esc_motors.md

For more information see:
* [PWM ESCs and Servos](../peripherals/pwm_escs_and_servo.md)
* [OneShot ESCs and Servos](../peripherals/oneshot.md)
* [DShot](../peripherals/dshot.md)
* [UAVCAN ESCs](../peripherals/uavcan_escs.md)
* [ESC Calibration](../advanced_config/esc_calibration.md)
* [ESC Firmware and Protocols Overview](https://oscarliang.com/esc-firmware-protocols/) (oscarliang.com)

A high level overview of the main ESC/Servo protocols supported by PX4 is given below.

## ESC Protocols

### PWM

[PWM ESCs](../peripherals/pwm_escs_and_servo.md) are commonly used for fixed wing vehicles and ground vehicles (vehicles that require a higher response rate like multicopters typically use oneshot or dshot ESCs).

PWM ESCs communicate using a periodic pulse, where the _width_ of the pulse indicates the desired power level.
The pulse wdith typically ranges between 1000uS for zero power and 2000uS for full power.
The periodic frame rate of the signal depends on the capability of the ESC, and commonly ranges between 50Hz and 490 Hz (the theoretical maximum being 500Hz for a very small "off" cycle).

> A higher rate is better, in particular where a rapid response to setpoint changes is needed.

Author: ```suggestion
```suggestion
A higher rate is better for ESCs, in particular where a rapid response to setpoint changes is needed.
For PWM servos 50Hz is usually sufficient, and many don't support higher rates.
``` 
```

> A higher rate is better, in particular where a rapid response to setpoint changes is needed.

Author: Will do! Thanks

There are a number of variants of the OneShot protocol, which support different rates.
PX4 only supports OneShot 125.

OneShot 125 is the same as PWM but uses pulse widths that are 8 times shorter (from 125us to 250us for zero to full power).
This allows OneShot 125 ESCs to have a much shorter duty cycle/higher rate. 
For PWM the theoretical maximum is close to 500 Hz while for OneShot it approaches 4 kHz.
The actual supported rate depends on the ESC used.

Setup:
- [ESC Wiring](../peripherals/pwm_escs_and_servo.md) (same as for PWM ESCs)
- [PX4 Configuration](../peripherals/oneshot.md#px4-configuration)
- [ESC Calibration](../advanced_config/esc_calibration.md)

### DShot

[DShot](../peripherals/dshot.md) is a digital ESC protocol that is highly recommended for vehicles that can benefit from reduce latency, in particular racing multicopters, VTOL vehicles, and so on.

It has reduced latency annd is more robust than both [PWM](#pwm) and [OneShot](#oneshot).
In addition it does not require ESC calibration, telemetry is available from some ESCs, and you can revers motor spin directions

> PX4 configuration is done by setting the desired rate .

Author: ```suggestion
PX4 configuration is done by setting the desired rate using [DSHOT_CONFIG](../advanced_config/parameter_reference.md#DSHOT_CONFIG).
```

> The parameter name indicates the data rate in kbaud - e.g. DShot150 has a rate of 150kbaud.

Author: ```suggestion
The parameter value indicates the data rate in kbaud - e.g. DShot150 has a rate of 150kbaud.
```

> Generally higher rates are better, but lower rates may be needed for larger aircraft (with longer leads).

Author: ```suggestion
Higher rates result in lower latency, but higher rates are more robust (and hence more suitable for large aircraft with longer leads); some ESCs only support lower rates (see datasheets for information).
```

Setup:
- [ESC Wiring](../peripherals/pwm_escs_and_servo.md) (same as for PWM ESCs)
- [PX4 Configuration](../peripherals/dshot.md#configuration) (using [DSHOT_CONFIG](../advanced_config/parameter_reference.md#DSHOT_CONFIG))
- [DShot](../peripherals/dshot.md) also contains information about how to send commands etc.

### UAVCAN

> [UAVCAN ESCs](../peripherals/uavcan_escs.md) are highly recommended when UAVCAN is the primary bus used for your vehicle.

Author: Thanks. Good to know. I will modify to include this information about rate.

```suggestion
[UAVCAN ESCs](../peripherals/uavcan_escs.md) are recommended when UAVCAN is the primary bus used for your vehicle.
The current PX4 implementation has an update rate of 200Hz.
```

### DShot

[DShot](../peripherals/dshot.md) is a digital ESC protocol that is highly recommended for vehicles that can benefit from reduce latency, in particular racing multicopters, VTOL vehicles, and so on.

It has reduced latency annd is more robust than both [PWM](#pwm) and [OneShot](#oneshot).
In addition it does not require ESC calibration, telemetry is available from some ESCs, and you can revers motor spin directions

PX4 configuration is done by setting the desired rate using [DSHOT_CONFIG](../advanced_config/parameter_reference.md#DSHOT_CONFIG).
The parameter value indicates the data rate in kbaud - e.g. DShot150 has a rate of 150kbaud.
Higher rates result in lower latency, but lower rates are more robust (and hence more suitable for large aircraft with longer leads); some ESCs only support lower rates (see datasheets for information).

Setup:
- [ESC Wiring](../peripherals/pwm_escs_and_servo.md) (same as for PWM ESCs)
- [PX4 Configuration](../peripherals/dshot.md#configuration) (using [DSHOT_CONFIG](../advanced_config/parameter_reference.md#DSHOT_CONFIG))
- [DShot](../peripherals/dshot.md) also contains information about how to send commands etc.

### UAVCAN

> [UAVCAN ESCs](../peripherals/uavcan_escs.md) are recommended when UAVCAN is the primary bus used for your vehicle.

Author: ```suggestion
[UAVCAN ESCs](../peripherals/uavcan_escs.md) are recommended when UAVCAN is the primary bus used for your vehicle.
The PX4 implementation is currently limited to update rates of 200Hz.
```
